### ISSUE TITLE
Vsync context relative, minimum version to high

### ISSUE THREAD

**Issue by boorishpea** [glfw/glfw#938]
I am adding multimonitor support to my application, as soon as I use more than one window I get screen tears, so I enabled vsync to find that it is context relative, which is alright however because I enable vsync for each windows context using the minimum value (1second) it is applied too both contexts which means that with two windows the FPS drops from 60fps (with one) to 30fps (with two windows)

Is there a reason why it is context relative?
Can the value be even lower? (i.e 0.5 so that both windows cap to 60fps)

**Comment by elmindreda** [comment]
A common workaround is to set the swap interval to zero for all windows except the first, and set the interval for that to one. That way you only wait once per update of all windows.

**Comment by boorishpea** [comment]
I find this to be a suitable solution. Maybe all that is required is for this to be mentioned in the documentation. Thank you.

**Comment by elmindreda** [comment]
Yeah, it really should be documented. I'll be adding a _Common tasks_ guide and this problem is [on the list](https://github.com/glfw/glfw/wiki/Common-tasks).

**Comment by linkmauve** [comment]
This is a bad idea imo, since if the two windows are on two monitors that have a different refresh rate you will be badly synchronised.

I would personally recommend to use one thread per window, that would just `glfwSwapBuffers()` and then either signal to the main loop that it can draw again, or handle drawing in the thread itself.  You have to take care of potentially huge differences in refresh rate though, 60fps isn’t the only one anymore, some older display controllers can only drive 4k monitors at 30fps, and 120fps or even 144fps monitors exist too.

**Comment by elmindreda** [comment]
I don't have the hardware to verify, but I suspect that OpenGL swap interval simply follows the primary monitor whatever other hardware is plugged in.

**Comment by paperManu** [comment]
@elmindreda AFAIK, you're right about this. The software I develop, which can output on many monitors (a project-mapping engine) is synced on the main display, and this is the reason we need to use professional cards with sync cards (like Quadros with Quadro Sync) to have synchronization on all ouptuts. And even with this, the best is to have projectors synced to.
